### PR TITLE
FIX: issue #4432 ([CRASH] involving Parse, Reduce & GC)

### PR DESCRIPTION
**Pull request by qtxie**
Those fixes are from PR #4097.  @hiiamboris 👍

### CODE REVIEW

File: runtime/datatypes/string.reds [hunk 1]
```diff
@@ -2741,10 +2751,11 @@ string: context [
 			type: TYPE_OF(cell)
 			either type = TYPE_CHAR [
 				char: as red-char! cell
+				s: GET_BUFFER(str)
 				either part? [				;-- /part will insert extra elements
-					s: insert-char s str/head + added char/value
+					insert-char s str/head + added char/value
 				][
-					s: overwrite-char s str/head + added char/value
+					overwrite-char s str/head + added char/value
```
Comment (dockimbel) [inline runtime/datatypes/string.reds:2758]: Why removing the `s:` part?

**Comment by greggirwin** [comment]
Saw a comment from @dockimbel about `s:` being removed. My guess is that it's because `s` wasn't used after that, and updating it leads you think it is.

**Comment by qtxie** [comment]
@greggirwin Yes. `s` wasn't used after that and we always fetch `s` before using it. So `s` always points to valid memory.

**Comment by dockimbel** [comment]
@qtxie @greggirwin I removed my comment a few minutes after posting it.

**Pull request merged**